unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -210,12 +278,128 @@ public async UniTask<FriendRequest> RequestFriendshipAsync(string friendUserId,
                         Message = messageBody,
                         User = new User() { Address = friendUserId },
                     }
+                },
+                AuthToken = new Payload()
+                {
+                    SynapseToken = accessToken
                 }
             };
 
-            await UpdateFriendship(updateFriendshipPayload, cancellationToken);
+            return await UpdateFriendship(updateFriendshipPayload, friendUserId, cancellationToken);
+        }
+
+        private async UniTask SubscribeToFriendshipEvents(CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Does it make sense to throw or you can just return if cancellation is requested?

Reply:
Actually i did a refactor of the cancellation token management here